src/vehicle.cpp

```diff
@@ -3494,6 +3456,24 @@ void vehicle::idle(bool on_map) {
             }
 
             if( on_map ) {
+                static std::map<int, std::string> opts = {
+                    {  15, _( "hummm!" ) },
+                    {  30, _( "whirrr!" ) },
+                    {  60, _( "vroom!" ) },
+                    { 100, _( "roarrr!" ) },
+                    { 140, _( "ROARRR!" ) },
+                    { 180, _( "BRRROARRR!!" ) },
+                    { INT_MAX, _( "BRUMBRUMBRUMBRUM!!!" ) }
+                };
+
+                int noise = engine_noise( eng, load );
+
+                if( noise > 0 ) {
+                    auto iter = opts.begin();
+                    for( ; iter != opts.end() && iter->first < noise; ++iter )
```

Comment:
It would spam with `sounds::ambient_sound()` because currently it looks like this:
```c++
for( ... ) {
    sound::..
}

```
E.g. if you have `noise` of 190, it would do all the `opt`s.